File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeBidPriceMessage.scala





















// Generated on 05/03/2013 10:34:53
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeBidPriceMessage extends MessageDeserializer {
  val messageId = 5755

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeBidPriceMessage.messageId

    val genericId: Int
    val averagePrice: Int

    def serialize(buf: Buffer) {
      buf.writeInt(genericId)
      buf.writeInt(averagePrice)
    }

  }

  protected def create(genericId: Int, averagePrice: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val genericId = buf.readInt()
    val averagePrice = buf.readInt()

    create(genericId, averagePrice)
  }

}

object ExchangeBidPriceMessage extends ExchangeBidPriceMessage {
  case class MessageImpl(genericId: Int, averagePrice: Int) extends Message
  type Target = MessageImpl

  override def create(genericId: Int, averagePrice: Int) = MessageImpl(genericId, averagePrice)

  def apply(genericId: Int, averagePrice: Int) = create(genericId, averagePrice)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeBidPriceMessage.java


// Generated on 05/08/2013 19:37:53
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeBidPriceMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5755;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int genericId;
    public int averagePrice;
    
    public ExchangeBidPriceMessage() { }
    
    public ExchangeBidPriceMessage(int genericId, int averagePrice) {
        this.genericId = genericId;
        this.averagePrice = averagePrice;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(genericId);
        buf.writeInt(averagePrice);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        genericId = buf.readInt();
        if (genericId < 0)
            throw new RuntimeException("Forbidden value on genericId = " + genericId + ", it doesn't respect the following condition : genericId < 0");
        averagePrice = buf.readInt();
    }
    
}
